doc/dev/tma.md

```diff
@@ -86,12 +90,87 @@ We call this split expression a "*boxing split*", the input of this split a "*pa
 the inner output of this split a "*box IterDomain*", and the outer output of this split a "*coordinate IterDomain*".
 
 For the case of Figure 1, if both box dimensions are defined by partitioning,
-the schedule should look like the Figure 3 below:
+the schedule should look like the Figure 4 below:
 
-![Figure 3: Boxing by partitioning](tma/box-by-partitioning.svg)
+![Figure 4: Boxing by partitioning](tma/box-by-partitioning.svg)
 
 Please note that, although in the above example, the split is divisible, this does not have to be the case in general.
 
 #### Define box by compositing
 
-TODO: this documentation is under construction
+Define box by compositing is to select a slice of IterDomains from the TMA domain and consider the innermost IterDomain(s) in the slice as box.
```

Comment:
If the `I5` TMA domain is scheduled, then can we replace this way of defining boxes by the above partitioning method? I wonder why we need this method in addition to the partitioning.

Reply:
Yes, we can always replace this way of defining boxes by partitioning. This way exists primarily for convenience. For example, if I have a discontiguous tensor [I1, I2], then you will be forced to partition both I1 and I2. But what often happen is, you only want to partition I2, and use box size 1 for I1. With this way of defining box, we will be able to automatically consider this as a size 1 box. Another example is, when you want to write a unit test, you often want to treat the entire dim as box, with only partitioning, you are forced to split this dim by its extent to get a [1, N]. But with this mechanism, you will be able to just parallelize this dim as `ParallelType::Bulk`.